jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/remote/RemoteLambdaRunner.kt

```diff
@@ -4,14 +4,39 @@
 package software.aws.toolkits.jetbrains.services.lambda.execution.remote
 
 import com.intellij.execution.configurations.RunProfile
+import com.intellij.execution.configurations.RunProfileState
+import com.intellij.execution.configurations.RunnerSettings
 import com.intellij.execution.executors.DefaultRunExecutor
-import com.intellij.execution.runners.DefaultProgramRunner
+import com.intellij.execution.runners.AsyncProgramRunner
+import com.intellij.execution.runners.ExecutionEnvironment
+import com.intellij.execution.runners.RunContentBuilder
+import com.intellij.execution.ui.RunContentDescriptor
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.ModalityState
+import com.intellij.openapi.application.runInEdt
+import org.jetbrains.concurrency.AsyncPromise
+import org.jetbrains.concurrency.Promise
 
-class RemoteLambdaRunner : DefaultProgramRunner() {
+class RemoteLambdaRunner : AsyncProgramRunner<RunnerSettings>() {
     override fun getRunnerId(): String = "Remote AWS Lambda"
 
-    override fun canRun(
-        executorId: String,
-        profile: RunProfile
-    ): Boolean = DefaultRunExecutor.EXECUTOR_ID == executorId && profile is RemoteLambdaRunConfiguration
+    override fun canRun(executorId: String, profile: RunProfile): Boolean =
+        DefaultRunExecutor.EXECUTOR_ID == executorId && profile is RemoteLambdaRunConfiguration
+
+    override fun execute(environment: ExecutionEnvironment, state: RunProfileState): Promise<RunContentDescriptor?> {
+        val runPromise = AsyncPromise<RunContentDescriptor?>()
+        ApplicationManager.getApplication().executeOnPooledThread {
+            val builder = state.execute(environment.executor, this)?.let {
+                RunContentBuilder(it, environment)
+            }
+
+            builder?.let {
```

Comment:
does the promise needs to be failed if the builder is null?

Reply:
Our code can't return null, we can probably downcast it to make it safe

Reply:
No longer possible if we downcast, and thats guarded by the canRun method so we good :D